ccm_web/server/src/api/api.controller.ts

```diff
@@ -1,8 +1,10 @@
 import { Controller, Get } from '@nestjs/common'
+import { ApiBearerAuth } from '@nestjs/swagger'
 
 import { HelloData, Globals } from './api.interfaces'
 import { APIService } from './api.service'
 
+@ApiBearerAuth()
```

Comment:
This seems to be doing the magic of recognizing if the path is from swagger and checking for token and passing or un passing. 

I commented out that line and try to do an api call from swagger throws error token required nicely! But uncommenting that line and making the api call gave me error as well. I tried to do an LTI launch again, picked the new  token etc. I had to restart the server so that controller recognize  `ApiBearerAuth`. This is ok, Not an issue. Just stating what I noticed.

Reply:
Thanks for looking closely at all this. My understanding is that the creation of the OpenAPI document relies a lot decorators, both those that are part of Nest and additional ones created for OpenAPI purposes. See [Nest - OpenAPI - Types and Parameters](https://docs.nestjs.com/openapi/types-and-parameters) and [Nest - OpenAPI - Decorators](https://docs.nestjs.com/openapi/decorators).